test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests.cs

```diff
@@ -119,20 +118,21 @@ public static void RegistersTelemetryConfigurationFactoryMethodThatReadsInstrume
             }
 
             /// <summary>
-            /// Tests that the Active configuration singleton is used as the telemetry configuration instance by the configuration factory.
-            /// This demonstrates that existing documentation for how to create a telemetry client and track custom events etc. works in ASP.NET 5
-            /// when no ApplicationInsights.config file exists but a project.json file does exist which contains the instrumentation key.
+            /// Tests that the Active configuration singleton is updated, but another instance of telemetry configuration is created for dependency injection.
+            /// ASP.NET Core developers should always use Dependency Injection instead of static singleton approach. 
```

Comment:
It's been a pattern for a while in AI SDKs to use `Active` for central overrides such as instrumentation key or channels. Should we update some documentation in response to this change to make sure it does not contradict the new behavior?

Reply:
The difference in the behavior is only noticeable when altering TelemetryConfiguration.Active after calling `AddApplicationInsights`.

Previously, adding TelemetryInitializers/Processors, modifying other settings would have an effect on the telemetry collected automatically by the modules: requests, depednencies, exceptions.

Now, it would not as TelemetryConfiguration used by SDK and Active are different objects.

I think it's just a best practice for DI-based systems to never use static singleton and always resolve services through the DI.

We also have documentation that shows how to properly do things in AspNetCore SDK here: https://docs.microsoft.com/en-us/azure/application-insights/app-insights-api-filtering-sampling

However still, it could be a breaking change as some of apps may follow other not-that-good practices.

Reply:
@lmolkova 
We have been advising people (incorrectly?) to get TelemetryConfiguration from Active, and modify it in some of the documentations.
Example below:
https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Custom-Configuration#adding-new-telemetryinitializer
"one can obtain the TelemetryConfiguration object (either from DI or by using TelemetryConfiguration.Active), and add new initializers."
The correct way to modify TelemetryConfiguration so that SDK modules also pick up changes is to get TelemetryConfiguration from DI, and modify it - is this correct? I can update the wiki docs to reflect this as the correct approach.

Reply:
If this change introduce potential issues for folks who relied on SDK modules picking up TC.Active modifications, lets mention that in Changelog.md itself so its easier to explain later :)

Reply:
@cijothomas 
It is correct that DI approach should be always preferred over static singletons in ASP.Net Core or any project where DI is used.  
I.e. 
`TelemetryConfiguration.Active.TelemetryInitializers.Add(new MyCustomTelemetryInitializer())`

should not be promoted as a way to add telemetry initializers.
The reasons are demonstrated by this bug: issues global state, hidden dependencies, poor testability.

And you are right, it introduces potential issues for those who used TC.Active. I'll update changelog with this info.

Reply:
Can you please look at this FAQ i added and make corrections if needed ?
https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/FAQs#2-accessing-telemetryconfiguration-via-telemetryconfigurationactive-vs-using-di

Reply:
@cijothomas looks great, thanks for changing docs!